app/models/concerns/order_shipping_method.rb

```diff
@@ -18,4 +18,17 @@ def shipping_method
     return if shipments.empty?
     shipments.first.shipping_method
   end
+
+  # Finds the shipment shipping_rate for the given shipping_method_id and selects that shipping_rate
+  #
+  # @return [Shipment]
+  def select_shipping_method(shipping_method_id)
+    return if shipments.empty?
+    shipment = shipments.first
+
+    shipping_rate = shipment.shipping_rates.find{ |shipping_rate| shipping_rate.shipping_method_id == shipping_method_id }
```

Comment:
Is this an ActiveRecord relation? Then we could do `shipment.shipping_rates.find_by_shipping_method_id(shipping_method_id)` and let the database handle this.

Reply:
And we can skip it if `shipping_method_id` is empty.